src/main/scala/com/atomist/rug/test/gherkin/handler/AbstractHandlerScenarioWorld.scala

```diff
@@ -57,16 +58,39 @@ abstract class AbstractHandlerScenarioWorld(definitions: Definitions, rugs: Opti
   /**
     * Return the plan or throw an exception if none was recorded
     */
-  def requiredPlan: jsPlan = {
+  def requiredPlan: jsScalaHidingProxy = {
     //println(s"Contents of recorded plan: $planOption")
-    planOption.map(new jsPlan(_)).getOrElse(throw new IllegalArgumentException("No plan was recorded"))
+    planOption.map(jsScalaHidingProxy.apply(_)).getOrElse(throw new IllegalArgumentException("No plan was recorded"))
   }
 
   /**
     * Return the plan or null if none was recorded
     */
-  def plan: jsPlan = {
-    planOption.map(new jsPlan(_)).orNull
+  def plan: jsScalaHidingProxy = {
+    planOption.map(jsScalaHidingProxy.apply(_)).orNull
+  }
+
+  /**
+    * Is the plan internally valid? Do the referenced handlers and other operations exist?
+    */
+  def planIsInternallyValid(): Boolean = {
+    val p = planOption.getOrElse(throw new IllegalArgumentException("No plan was recorded"))
+    !p.instructions.map(_.instruction).exists {
+      case Edit(detail) =>
+        val knownEditors: Seq[String] = rugs.map(_.editorNames).getOrElse(Nil)
+        !knownEditors.contains(detail.name)
+      case Generate(detail) =>
+        val knownGenerators: Seq[String] = rugs.map(_.generatorNames).getOrElse(Nil)
+        !knownGenerators.contains(detail.name)
+      case Review(detail) =>
+        val knownReviewers: Seq[String] = rugs.map(_.reviewerNames).getOrElse(Nil)
+        !knownReviewers.contains(detail.name)
+      case Command(detail) =>
+        val knownCommandHandlers: Seq[String] = rugs.map(_.commandHandlerNames).getOrElse(Nil)
+        !knownCommandHandlers.contains(detail.name)
+        // TODO there are probably more cases here
+      case _ => false
+    }
```

Comment:
Thinking a bit more about this, it seems you are only checking what you know you can check. If it is something you can't check, you are allowing it.